lib/matplotlib/lines.py

```diff
@@ -688,7 +696,11 @@ def recache(self, always=False):
             interpolation_steps = self._path._interpolation_steps
         else:
             interpolation_steps = 1
-        xy = STEP_LOOKUP_MAP[self._drawstyle](*self._xy.T)
+
+        if self._drawstyle in ["steps-between", "steps-edges"]:
+            xy = STEP_LOOKUP_MAP[self._drawstyle](*self._xy)
```

Comment:
Is this because these functions expect a different signature or because we are storing the data different?  I don't fully understand what is going on here, but it is making me a bit nervous...

Reply:
It's related to the workaround in L671-L677. The current setup expects xy to be Nx2, so I patched it to account for the len(x) + 1 = len(y), but I didn't want to change the return type overall to not break something. Happy to change it, if you can think of a better solution.